src/commands/config/setCluster.ts

```diff
@@ -0,0 +1,233 @@
+import { cliContainer } from "../../container";
+import { ClusterConfig } from "../../types";
+
+interface ClusterOption {
+  displayName: string;
+  debug: boolean;
+  cluster?: ClusterDef;
+}
+
+interface ClusterDef {
+  name: string;
+  apiKey: string;
+  endpoint: string;
+}
+
+const clusterOptions: ClusterOption[] = [
+  {
+    displayName: "skygeario",
+    debug: false,
+    cluster: {
+      name: "skygeario",
+      apiKey: "sq3GcUp0QVRTYPwBpnLcQi3jHK7SJFfF",
+      endpoint: "https://controller.skygear.dev",
+    },
+  },
+  {
+    displayName: "skygeario staging",
+    debug: true,
+    cluster: {
+      name: "skygeario-staging",
+      apiKey: "nA1nZ2vemgjJFD99n36QoGNObG5myWXO",
+      endpoint: "https://controller.staging.skygear.dev",
+    },
+  },
+  {
+    displayName: "skygeario dev",
+    debug: true,
+    cluster: {
+      name: "skygeario-dev",
+      apiKey: "3ac91f63e599eafce8632420613754cb",
+      endpoint: "https://controller.dev.skygearapis.com",
+    },
+  },
+  {
+    displayName: "Connect to my own cluster",
+    debug: false,
+  },
+];
+
+const apiKeyPrompt: inquirer.Question = {
+  message: "Cluster API key:",
+  name: "apiKey",
+  type: "input",
+  validate: input => {
+    if (input.trim() === "") {
+      return "Cluster API key is required.";
+    }
+    return true;
+  },
+};
+
+const urlPrompt: inquirer.Question = {
+  message: "Cluster endpoint:",
+  name: "endpoint",
+  type: "input",
+  validate: input => {
+    if (input.trim() === "") {
+      return "Cluster API endpoint is required.";
+    }
+    return true;
+  },
+};
+
+const namePrompt: inquirer.Question = {
+  message: "Cluster Name:",
+  name: "cluster",
+  type: "input",
+  validate: input => {
+    if (!/^[a-zA-Z0-9-]{1,}$/.test(input)) {
+      return "Cluster Name must contain only alphanumeric characters and dash.";
+    }
+    return true;
+  },
+};
+
+async function selectCluster(argv: Arguments) {
+  const answers = await inquirer.prompt([
+    {
+      choices: clusterOptions
+        .filter(o => !o.debug || argv.debug)
+        .map(t => ({ name: t.displayName, value: t.cluster })),
+      message: "Select a cluster you want to connect to:",
+      name: "cluster",
+      type: "list",
+    },
+  ]);
+  return answers.cluster as ClusterDef | undefined;
+}
+
+async function askClusterServer(argv: Arguments) {
+  const prompts = [];
+  const server: ClusterDef = {
+    name: argv.cluster as string,
+    endpoint: argv.endpoint as string,
+    apiKey: argv["api-key"] as string,
+  };
+  if (!server.endpoint || !server.apiKey) {
+    let selected: ClusterDef | undefined;
+    if (!server.name) {
+      selected = await selectCluster(argv);
+      if (selected) {
+        // eslint-disable-next-line require-atomic-updates
+        server.name = selected.name;
+      }
+    } else {
+      selected = clusterOptions.find(c => c.cluster?.name === server.name)
+        ?.cluster;
+    }
+
+    if (!selected) {
```

Comment:
From cluster selection prompt in `selectCluster`, we have option `Connect to my own cluster` that cluster is `undefined`. If user select this option, the `selected` will become `undefined` and error will be shown. For this case, we should show prompt to ask user to input endpoint and api key instead?